### PR TITLE
PLAT-68568: added send eta to copilot apps

### PR DESCRIPTION
**Pull request by haileyR**

### CODE REVIEW

File: console/src/App/AppContextProvider.js [hunk 1]
```diff
@@ -156,17 +156,17 @@ class AppContextProvider extends Component {
...
-					this.setWeather(this.state.location.lat, this.state.location.lon);
-				} else {
-					// Just use location services.
-					this.updateAppState((state) => {
-						state.location.lat = position.coords.latitude;
-						state.location.lon = position.coords.longitude;
-					});
-					this.setWeather(position.coords.latitude, position.coords.longitude);
-				}
+				// if (this.state.connections.serviceLayer) {
```
[the remark this answers is not in the document]
Comment (haileyR) [reply console/src/App/AppContextProvider.js]: I thought we are commenting this out for now.

[the remark this answers is not in the document]
Comment (Djspaceg) [reply console/src/App/AppContextProvider.js]: Nah, just on your screen for development. We still need the location to be updated from the simulator's location in general. It's just that the simulator comes online tomorrow and isn't sending out location data right now.

File: console/src/components/MapCore/MapCore.js [hunk 2]
```diff
@@ -593,11 +592,12 @@ const SkinnableMap = AppContextConnect(({location, userSettings, updateAppState}
 	location,
 	// destination: navigation.destination,
 	updateNavigation: ({duration}) => {
+		const now = new Date().getTime();
+		const eta = new Date(now + (duration * 1000)).getTime();
 		updateAppState((state) => {
-			const now = new Date().getTime();
 			state.navigation.duration = duration;
 			state.navigation.startTime = now;
-			state.navigation.eta = new Date(now + (duration * 60000)).getTime();
```
Comment (Djspaceg) [inline console/src/components/MapCore/MapCore.js]: :lol: thanks for catching this 60000 -> 1000 bug. I was WAY OFF!

**Comment by Djspaceg** [comment]
Stumbled across a way, that I should have realized before now, which allows the ServiceLayer to just automatically respond to changes in the app state, making everything like this so much simpler, so this PR was a HUGE HELP. App state comes in as props via AppContextProvider which we can then just compare for differences, acting on anything that's different like a normal component (duh). This is what can fire the internal methods which communicate with the other app. I've made the change already to the ETA->navigation object, and we can probably even convert the setting of the destination to this technique too; we can save that for another PR.
Also fancied-up the duration countdown display.

**Pull request merged**